### PR TITLE
Downgrade Python to 3.8

### PR DESCRIPTION
**Pull request by iaindillingham**
This matches the version on the OpenSAFELY Python Docker image. We let dependabot know by adding .python-version. We also update the dev and prod requirements for good measure.

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.8.10
```
Comment (ghickman) [inline .python-version:1]: This doesn't match the point release in python-docker's .python-version, is that expected?

File: requirements.dev.txt
Comment (ghickman) [inline requirements.dev.txt:1]: Did you mean to update various dependency versions in this file?

**Review by ghickman — APPROVED** [review]

File: .python-version (hunk 1 above)
Comment (iaindillingham) [reply .python-version:1]: It isn't, but I think that python-docker's .python-version is at fault.

```sh
opensafely exec python --version
```

returns `Python 3.8.10`.

File: requirements.dev.txt
Comment (iaindillingham) [reply requirements.dev.txt:1]: Yes, although "for good measure" is vague 🙂 I wanted to ensure that dependencies were 3.8-ready, rather than 3.10-ready.

**Pull request merged**